src/rules/vue-exports-component-directive.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const path = require( 'upath' );
+
+/**
+ * Vue files should have `// @vue/component` before their
+ * module.exports so that rules from eslint-plugin-vue are
+ * triggered.
+ *
+ * @author DannyS712
+ */
+module.exports = {
+	meta: {
+		type: 'suggestion',
+		docs: {
+			description: 'Require `// @vue/component` directives to trigger eslint-plugin-vue rules'
+		},
+		schema: [],
+		messages: {
+			'missing-directive': 'The `// @vue/component` directive should be included on the line before module.exports'
+		}
+	},
+	create( context ) {
+		// Don't trigger on files that don't end in .vue
+		const fileName = context.getFilename();
+		const fileExt = path.extname( fileName );
+		if ( fileExt !== '.vue' ) {
+			// No rules
+			return {};
+		}
+
+		// Must be a Vue file, look for `module.exports =`
+		return {
+			AssignmentExpression: ( node ) => {
+				// Early return if its not `module.exports =`
+				if ( node.operator !== '=' ||
```

Comment:
I suggest adding a check for `node.type !== 'AssignmentExpression'` (or whatever the correct type is), so that we're sure that node.left exists and we won't crash trying to access `node.left.type`. In practice, this is probably OK because I don't think there's any way `node` could have `.operator === '='` without also having a `.left` property, but I think it would be good to be explicit about what type of node we're looking at.

Reply:
Isn't this already enforced by the fact that this should only be triggered for AssignmentExpression ? Happy to add, but my understanding of the documentation I read and examples I saw is that the fact that the object key here is AssignmentExpression means that the node must be that type

Reply:
Oh, duh, you're absolutely right. I missed the "AssignmentExpression" thing above.

Reply:
done

Reply:
Edit conflict - do you want me to remove it?

Reply:
Yes, we'll need to remove it because it causes the test coverage to be below 100%, since line 37 is never reached. Sorry, I'm an idiot.

Reply:
Done